src/ProjectTask.php

```diff
@@ -583,315 +584,45 @@ public static function getAllTicketsForProject($ID)
      **/
     public function showForm($ID, array $options = [])
     {
-        global $CFG_GLPI;
-
-        $rand_template           = mt_rand();
-        $rand_name               = mt_rand();
-        $rand_description        = mt_rand();
-        $rand_comment            = mt_rand();
-        $rand_project            = mt_rand();
-        $rand_state              = mt_rand();
-        $rand_type               = mt_rand();
-        $rand_percent            = mt_rand();
-        $rand_milestone          = mt_rand();
-        $rand_plan_start_date    = mt_rand();
-        $rand_plan_end_date      = mt_rand();
-        $rand_real_start_date    = mt_rand();
-        $rand_real_end_date      = mt_rand();
-        $rand_effective_duration = mt_rand();
-        $rand_planned_duration   = mt_rand();
-
         if ($ID > 0) {
+            $duration = ProjectTask_Ticket::getTicketsTotalActionTime($this->getID());
```

Comment:
Should be placed after next line, to save a useless DB query if check does not pass.